src/logic/currencyValues/store/reducer/currencyValues.js

```diff
@@ -4,27 +4,27 @@ import { type ActionType, handleActions } from 'redux-actions'
 
 import { SET_CURRENCY_BALANCES } from '~/logic/currencyValues/store/actions/setCurrencyBalances'
 import { SET_CURRENCY_RATE } from '~/logic/currencyValues/store/actions/setCurrencyRate'
-import { SET_CURRENT_CURRENCY } from '~/logic/currencyValues/store/actions/setCurrencySelected'
+import { SET_CURRENT_CURRENCY } from '~/logic/currencyValues/store/actions/setSelectedCurrency'
 import type { State } from '~/logic/tokens/store/reducer/tokens'
 
 export const CURRENCY_VALUES_KEY = 'currencyValues'
 
 export default handleActions<State, *>(
   {
     [SET_CURRENCY_RATE]: (state: State, action: ActionType<Function>): State => {
-      const { currencyRate } = action.payload
+      const { currencyRate, safeAddress } = action.payload
 
-      return state.set('currencyRate', currencyRate)
+      return state.setIn(['currencyValues', safeAddress, 'currencyRate'], currencyRate)
```

Comment:
this is duplicating the `currencyValues` entry in the store.

```suggestion
      return state.setIn([safeAddress, 'currencyRate'], currencyRate)
```